lib/sonice/public/script.js

```diff
@@ -24,6 +29,71 @@ $.fn.keyboardShortcut = function() {
   })
 }
 
+/* So nice helpers */
+
+// Get a new artist image from Last.fm via jsonp
+// When found calls the `callback` with the image url as the first argument
+function artistImage(artist, callback) {
+  if (!artist) return callback();
+  cache = artistImage.cache;
+  artist = encodeURI(artist);
+  const getImageFromCache = 
+    function() { callback(cache[artist].random()); };
+  
+
+  // Deliver from cache
+  if (cache.hasOwnProperty(artist)) {
+    // execute the callback asynchronously to minimize codepaths
+    setTimeout(getImageFromCache, 10);
+    return;
+  }
+
+  // Load
+  const lastFmUri = 'http://ws.audioscrobbler.com/2.0/?format=json&method=artist.getinfo&artist=%s&api_key=5636ca9fea36d0323a76638385aab1f3';
+  $.ajax({
+    url: lastFmUri.replace('%s', artist),
+    dataType: 'jsonp',
+    success: response => { 
+      //thanks to @hugovk! 
+      //(https://github.com/hugovk/now-playing-radiator)
+      const mbid = response.artist.mbid;
+      if (mbid) { 
```

Comment:
One way to simplify a bit the code here would be to replace this condition surrounding all of the next code by an early return.

```js
  …
  
  if (…) {
     …
     …
     …
  }
```

This can be transformed for less indentation with:

```js
  …
  
  if (!…) return;
  
  …
  …
  …
```